pinecone/runtime/src/main/java/io/quarkiverse/langchain4j/pinecone/PineconeEmbeddingStore.java

```diff
@@ -199,4 +206,20 @@ private void addAllInternal(List<String> ids, List<Embedding> embeddings, List<T
         Log.debug("Added embeddings: " + response.getUpsertedCount());
     }
 
+    private void waitForIndexToBecomeReady(String indexName, Duration timeout) {
```

Comment:
Does smallrye fault tolerance have any utils we can use here?

Reply:
Nothing coming to my mind, it's just a cycle that asks for the status until the status is ready.... 
At most we could use some retry logic on the individual REST client calls (which might be applicable to the whole client api, but we'd have to be very careful with retrying non-idempotent operations...)

Reply:
👌🏼